parsons/ngpvan/email.py

```diff
@@ -0,0 +1,143 @@
+from parsons.etl.table import Table
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class Email(object):
+    def __init__(self, van_connection):
+
+        self.connection = van_connection
+
+    def get_emails(
+        self,
+        orderby=None,
+    ):
+        """
+        Get emails.
+
+        Note that it takes some time for the system to aggregate opens and click-throughs,
+        so data can be delayed up to 15 minutes.
+
+        `Args:`
+            orderby : str
+                Optional; sorts results in ascending (asc) or descending (desc) order
+                for the dateModified field.
+
+        `Returns:`
+            Parsons Table
+                See :ref:`parsons-table` for output options.
+        """
+
+        params = {
+            "$orderby": orderby,
+        }
+
+        tbl = Table(self.connection.get_request("email/messages", params=params))
+        logger.debug(f"Found {tbl.num_rows} emails.")
+        return tbl
+
+    def get_email(self, email_id, expand=True):
+        """
+        Get an email.
+
+        `Args:`
+            email_id : int
+                The email id.
+            expand : bool
+                Optional; expands the email message to include the email content and
+                statistics. Defaults to True.
+
+        `Returns:`
+            dict
+        """
+
+        params = {
+            "$expand": (
+                "emailMessageContent, EmailMessageContentDistributions"
+                if expand
+                else None
+            ),
+        }
+
+        r = self.connection.get_request(f"email/message/{email_id}", params=params)
+        logger.debug(f"Found email {email_id}.")
+        return r
+
+    def get_email_stats(self):
+        """
+        Get stats for all emails.
+
+        `Args:`
+            emails : list
+                A list of email message details.
+
```

Comment:
Suggestion - maybe we could include `start_date` and `end_date` as optional parameters? I know that this won't affect `get_emails` but maybe might save some time for the L88 loop.

Reply:
Also, do we care about passing through `orderby`?